mergify_engine/queue/merge_train.py

```diff
@@ -340,6 +340,10 @@ async def update_user_pull(self, queue_rule: rules.QueueRule) -> None:
             await self._set_creation_failure(f"{exc.title}\n\n{exc.message}", "update")
             raise TrainCarPullRequestCreationFailure(self) from exc
 
+        # NOTE(sileht): We must update head_sha of the pull request otherwise next temporary pull request
+        # may be created on a vanish reference.
```

Comment:
```suggestion
        # NOTE(sileht): We must update head_sha of the pull request; otherwise next temporary pull request
        # may be created on a vanished reference.
```